test/junit/scala/collection/IteratorTest.scala

```diff
@@ -357,4 +357,32 @@ class IteratorTest {
     assertTrue(hi.hasNext) // no longer delegated
     assertTrue(hi.hasNext)
   }
+  @Test def `flatMap is memory efficient in previous element`(): Unit = {
+    import java.lang.ref._
+    // Array.iterator holds onto array reference; by contrast, iterating over List walks tail.
+    // Avoid reaching seq1 through test class.
+    val seq1 = new WeakReference(Array("first", "second"))
```

Comment:
if the array is only stored in a `WeakReference`, and not assigned elsewhere, doesn't that make it eligible for collection immediately?

Reply:
in particular, is `case 1 => seq1.get` not liable to sometimes return `null`?